redux/shoppersSlice.ts

```diff
@@ -54,6 +54,15 @@ export const shopperslice = createSlice({
         item!.quantity--;
       }
     },
+    updateItem: (state, action) => {
+      const itemIndex = state.productData.findIndex(
+        (item: StoreProduct) => item.id === action.payload._id
+      );
+      state.productData[itemIndex] = {
+        ...state.productData[itemIndex],
+        ...action.payload,
+      }
+    },
```

Comment:
Not sure if this is necessary, but I noticed you were trying to modify an "Item" in a map. With redux you should only ever modify a state object within the reducers here (not within the actual React component)

Reply:
Gotcha - still learning Redux but that is good to know. I'll try to be aware of this and not modify states directly from a component.